include/rocksdb/customizable.h

```diff
@@ -190,6 +190,20 @@ class Customizable : public Configurable {
   virtual const Customizable* Inner() const { return nullptr; }
 
  protected:
+  // Generates a unique ID for this instance of the customizable.
+  // The unique ID is of the form <name>:<addr>#pid, where:
+  // - name is the Name() of this object;
+  // - addr is the memory address of this object;
+  // - pid is the process ID of this process ID for this process.
+  // Note that if obj1 and obj2 have the same unique IDs, they must be the
+  // same.  However, if an object is deleted and recreated, it may have the
+  // same unique ID as a predecessor
+  //
+  // This method is useful for objects (especially ManagedObjects) that
+  // wish to generate an ID that is unique for this instance and wish to
+  // override the GetId() method.
+  std::string GenerateUniqueId() const;
```

Comment:
I think calling this a "unique ID" is confusing in comparison to other APIs (unique_id.h, Env::GenerateUniqueId()). And Generate suggests it's new each time.

How about `GetInstanceId()`?